web/courses_new.go

```diff
@@ -9,127 +9,42 @@ import (
 	"github.com/quickfeed/quickfeed/scm"
 )
 
-const (
-	private = true
-	public  = !private
-)
-
-// RepoPaths maps from QuickFeed repository path names to a boolean indicating
-// whether or not the repository should be create as public or private.
-var RepoPaths = map[string]bool{
-	qf.InfoRepo:        public,
-	qf.AssignmentsRepo: private,
-	qf.TestsRepo:       private,
-}
-
 // createCourse creates a new course for the directory specified in the request
 // and creates the repositories for the course. Requires that the directory
 // does not contain the QuickFeed repositories that will be created.
 func (s *QuickFeedService) createCourse(ctx context.Context, sc scm.SCM, request *qf.Course) (*qf.Course, error) {
-	org, err := sc.GetOrganization(ctx, &scm.GetOrgOptions{ID: request.OrganizationID})
+	courseCreator, err := s.db.GetUser(request.GetCourseCreatorID())
 	if err != nil {
-		return nil, err
-	}
-	if org.GetPaymentPlan() == FreeOrgPlan {
-		return nil, ErrFreePlan
+		return nil, fmt.Errorf("createCourse: failed to get course creator record from database: %w", err)
 	}
-	repos, err := sc.GetRepositories(ctx, org)
+	repos, err := sc.CreateCourse(ctx, &scm.NewCourseOptions{
+		CourseCreator:  courseCreator.Login,
+		OrganizationID: request.OrganizationID,
+	})
 	if err != nil {
+		s.logger.Debugf("createCourse: failed to create course repositories or teams: %w", err)
```

Comment:
Should not log error messages that also get returned up the stack. Log only once.

Reply:
%w does not work in log messages, only in errors